renovate.json

```diff
@@ -12,28 +12,80 @@
     {
       "matchManagers": ["poetry", "pip_requirements", "regex"],
       "matchDatasources": ["pypi"],
+      "excludePackageNames": ["juju", "libjuju 2", "libjuju 3"],
       "groupName": "Python dependencies"
     }, {
       "matchManagers": ["github-actions"],
       "groupName": "GitHub actions"
     }, {
+      "matchDepNames": ["Juju 3"],
+      "matchPackageNames": ["juju/juju"],
+      "allowedVersions": "<3.2.0",
+      "extractVersion": "^juju-(?<version>.*)$",
+      "groupName": "Juju 3"
+    }, {
+      "matchDepNames": ["Juju 2"],
       "matchPackageNames": ["juju/juju"],
       "allowedVersions": "<3.0.0",
       "extractVersion": "^juju-(?<version>.*)$",
-      "groupName": "Juju agent"
+      "groupName": "Juju 2"
+    }, {
+      "matchDepNames": ["libjuju 3"],
+      "matchPackageNames": ["juju"],
+      "matchDatasources": ["pypi"],
+      "allowedVersions": "<3.2.0",
+      "groupName": "Juju 3"
     }, {
+      "matchDepNames": ["libjuju 2"],
       "matchPackageNames": ["juju"],
-      "allowedVersions": "<3.0.0"
+      "matchDatasources": ["pypi"],
+      "allowedVersions": "<3.0.0",
+      "groupName": "Juju 2"
```

Comment:
This renovate magic will not fully work, renovate will try to do major version bump for libjuju 2, but if we close the PR it shouldn't nag us further about it. Regular version updates should work as intended.